### PR TITLE
Validate phone number with international codes

### PR DESCRIPTION
**Pull request by padv2010**
## Trello Card Link

https://trello.com/c/PV6vpXTU/61-application-form-phone-number

## Description
The [design system for GOV.UK for phone numbers](https://design-system.service.gov.uk/patterns/telephone-numbers/) is still not conclusive when it 
comes to mobile numbers, but it has some experimental guidelines that I have
followed on this PR.

For the validations, at the time of writing this commit, the design system suggests 
[using a Library from Google named libphonenumber](https://github.com/google/libphonenumber). There is no implementation for
Ruby, but they point you towards one of the following:

1. [GitHub - ianks/mini_phone](https://github.com/ianks/mini_phone): A fast phone number parser, validator and formatter 
for Ruby. This gem binds to Google’s C++ libphonenumber for spec-compliance 
and performance.
3. [GitHub - daddyz/phonelib](https://github.com/daddyz/phonelib): Ruby gem for phone validation and formatting 
4. using google libphonenumber library data
5. [GitHub - mobi/telephone_number](https://github.com/mobi/telephone_number): Phone number validation for Ruby

We have chosen  [GitHub - daddyz/phonelib](https://github.com/daddyz/phonelib) as this one seems more stable and maintained. 

### Notes

1. As per the guidelines: `Let users enter telephone numbers in whatever format 
is familiar to them. Allow for additional spaces, hyphens, dashes and brackets, and 
be able to accommodate country and area codes.`
3. We use the validation via a Ruby extension of Google Library
4. We set the default telephone numbers to `GB` via an initializer. Not a fan of how it looks, but I can't think of a better way other than adding it to `application.rb`

### CODE REVIEW

File: spec/support/phone_number_validations.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+RSpec.shared_examples "validates phone number with international prefix" do |klass, attribute|
```
Comment (duncanjbrown) [inline spec/support/phone_number_validations.rb:3]: love this test

File: spec/models/applicants/personal_detail_spec.rb [hunk 2]
```diff
@@ -27,28 +27,36 @@ module Applicants
...
+
       describe "date_of_birth" do
         context "when date_of_birth is invalid" do
           let(:params) { { day: "31", month: "02", year: "2000" } }
 
           it "is invalid" do
-            expect(model).not_to be_valid
+            model.valid?
+
+            expect(model.errors["date_of_birth"]).not_to be_blank
```
Comment (duncanjbrown) [inline spec/models/applicants/personal_detail_spec.rb:39]: are these part of the same change?

**Review by duncanjbrown — APPROVED** [review]

File: spec/models/applicants/personal_detail_spec.rb (hunk 2 above)
Comment (padv2010) [reply spec/models/applicants/personal_detail_spec.rb:40]: @duncanjbrown I have edit the commit and replaced it with 833d3c7.

**Comment by padv2010** [comment]
@duncanjbrown I am merging this PR as it was approved and your comment implied a minor change (move content out of a commit).

**Pull request merged**